### PR TITLE
remove tensorflow backend in ops

### PR DESCRIPTION
**Pull request by Vivek305**
Remove tensorflow backend dependency from the Ops. Also modify the test scripts.

## Summary by CodeRabbit

- **Refactor**
  - Removed all TensorFlow support from tensor operations, models, and augmentations. The framework now exclusively supports PyTorch for all tensor operations and model handling.
  - Type annotations and internal logic were updated to use only PyTorch tensors and models.
- **Tests**
  - Removed all TensorFlow-based test cases and data from integration and unit tests.
  - Updated tests to use PyTorch tensors and models exclusively, ensuring continued coverage for PyTorch workflows.

### CODE REVIEW

File: fastestimator/op/numpyop/numpyop.py [hunk 1]
```diff
@@ -185,7 +184,7 @@ def __init__(self,
     def forward(self, data: Union[np.ndarray, List[np.ndarray]], state: Dict[str, Any]) -> None:
         pass
 
-    def forward_batch(self, data: Union[Tensor, List[Tensor]], state: Dict[str, Any]) -> None:
+    def forward_batch(self, data: Union[torch.Tensor, List[torch.Tensor]], state: Dict[str, Any]) -> None:
```
Comment (DwijayDS) [inline fastestimator/op/numpyop/numpyop.py:189]: We should keep `Tensor = TypeVar('Tensor', torch.Tensor, np.ndarray)`. and in the type use Union[Tensor, List[Tensor]]. Because in NumpyOp, it can stipp have np array

File: fastestimator/op/numpyop/numpyop.py [hunk 2]
```diff
@@ -223,8 +223,8 @@ def set_rua_level(self, magnitude_coef: float) -> None:
     def forward(self, data: List[np.ndarray], state: Dict[str, Any]) -> Union[np.ndarray, List[np.ndarray]]:
         return self.fn(*data)
 
-    def forward_batch(self, data: Union[Tensor, List[Tensor]], state: Dict[str,
-                                                                           Any]) -> Union[np.ndarray, List[np.ndarray]]:
+    def forward_batch(self, data: Union[torch.Tensor, List[torch.Tensor]],
+                      state: Dict[str, Any]) -> Union[np.ndarray, List[np.ndarray]]:
```
Comment (DwijayDS) [inline fastestimator/op/numpyop/numpyop.py]: We should keep this as `data: Union[Tensor, List[Tensor]]` because it could be np array

File: fastestimator/op/numpyop/numpyop.py [hunk 3]
```diff
@@ -264,7 +265,8 @@ def forward(self, data: List[np.ndarray], state: Dict[str, Any]) -> Optional[Fil
             return FilteredData(replacement=self.replacement)
         return None
 
-    def forward_batch(self, data: Union[Tensor, List[Tensor]], state: Dict[str, Any]) -> Optional[FilteredData]:
+    def forward_batch(self, data: Union[torch.Tensor, List[torch.Tensor]], state: Dict[str,
```
Comment (DwijayDS) [inline fastestimator/op/numpyop/numpyop.py]: We should keep this as data: Union[Tensor, List[Tensor]] because it could be np array

File: fastestimator/op/tensorop/augmentation/mixup_batch.py [hunk 4]
```diff
@@ -61,15 +60,13 @@ def __init__(self,
         self.shared_beta = shared_beta
 
     def build(self, framework: str, device: Optional[torch.device] = None) -> None:
-        if framework == 'tf':
-            self.beta = tfp.distributions.Beta(self.alpha, self.alpha)
-        elif framework == 'torch':
+        if framework == 'torch':
```
Comment (DwijayDS) [inline fastestimator/op/tensorop/augmentation/mixup_batch.py]: Do we need this `framework` parameter now?

File: fastestimator/op/tensorop/gradient/gradient.py [hunk 5]
```diff
@@ -80,12 +79,7 @@ def forward(self, data: List[Tensor], state: Dict[str, Any]) -> List[Tensor]:
                 results.append(get_gradient(final, initial, tape=state['tape'], retain_graph=retain_graph))
         else:
             finals = data
-            if self.framework == "tf":
-                trainable_params = self.model.trainable_variables
-                for idx, final in enumerate(finals):
-                    gradient = get_gradient(final, trainable_params, tape=state['tape'])
-                    results.append(gradient)
-            elif self.framework == "torch":
+            if self.framework == "torch":
```
Comment (DwijayDS) [inline fastestimator/op/tensorop/gradient/gradient.py]: Do we need this check?  We have assert statement to check framework on line 57

File: fastestimator/op/tensorop/loss/dice_loss.py [hunk 6]
```diff
@@ -80,10 +79,7 @@ def __init__(self,
                 self.weights[0, channel] = weight
 
     def build(self, framework: str, device: Optional[torch.device] = None) -> None:
-        if framework == 'tf':
-            if self.weights is not None:
-                self.weights = convert_tensor_precision(to_tensor(self.weights, 'tf'))
-        elif framework == 'torch':
+        if framework == 'torch':
```
Comment (DwijayDS) [inline fastestimator/op/tensorop/loss/dice_loss.py]: We can stop using the parameter framework and throw a depreciated warning for framework

File: fastestimator/op/tensorop/loss/dice_loss.py [hunk 7]
```diff
@@ -15,15 +15,14 @@
...
-import tensorflow as tf
 import torch
 
 from fastestimator.backend._convert_tensor_precision import convert_tensor_precision
 from fastestimator.backend._dice_score import dice_score
 from fastestimator.backend._to_tensor import to_tensor
 from fastestimator.op.tensorop.loss.loss import LossOp
 
-Tensor = TypeVar('Tensor', tf.Tensor, torch.Tensor, np.array)
+Tensor = TypeVar('Tensor', torch.Tensor, np.array)
```
Comment (DwijayDS) [inline fastestimator/op/tensorop/loss/dice_loss.py:25]: Not used in the file. We can remove it.

File: fastestimator/op/tensorop/loss/l2_regularization.py [hunk 8]
```diff
@@ -37,17 +34,18 @@ class L2Regularizaton(TensorOp):
         model: A tensorflow or pytorch model
```
Comment (DwijayDS) [inline fastestimator/op/tensorop/loss/l2_regularization.py]: We need to update this comment

File: fastestimator/op/tensorop/loss/super_loss.py [hunk 9]
```diff
@@ -80,29 +78,7 @@ def __init__(self,
 
     def build(self, framework: str, device: Optional[torch.device] = None) -> None:
```
Comment (DwijayDS) [inline fastestimator/op/tensorop/loss/super_loss.py:79]: We can ignore the framework parameter and throw depritiated warning and not condition on framework

File: test/PR_test/integration_test/op/numpyop/meta/test_fuse.py [hunk 10]
```diff
@@ -17,35 +17,7 @@ def __init__(self, inputs, outputs, mode, var):
...
-        save_path = tempfile.mkdtemp()
-        system.save_state(save_path)
-
-        # reinstantiate system and load the state
-        system = instantiate_system()
-        system.load_state(save_path)
-        loaded_var = system.pipeline.ops[0].ops[0].var
-
-        self.assertEqual(loaded_var, new_var)
+        cls.tf_data = torch.tensor([[1., 2., 4.], [1., 2., 6.]])
```
Comment (aashokvardhan) [inline test/PR_test/integration_test/op/numpyop/meta/test_fuse.py]: We can remove setup class since we no longer use this variable.

**Review by aashokvardhan — COMMENTED** [review]
there are minor changes

**Comment by DwijayDS** [comment]
@coderabbitai review

**Review by aashokvardhan — APPROVED** [review]
